filter_plugins/oo_filters.py

```diff
@@ -327,6 +329,54 @@ def oo_parse_heat_stack_outputs(data):
 
         return revamped_outputs
 
+    @staticmethod
+    def oo_parse_certificate_names(certificates, data_dir):
+        ''' Parses names from list of certificate hashes.
+
+            Ex: certificates = [{ "certfile": "/etc/origin/master/custom1.crt",
+                                  "keyfile": "/etc/origin/master/custom1.key" },
+                                { "certfile": "custom2.crt",
+                                  "keyfile": "custom2.key" }]
+
+                returns [{ "certfile": "/etc/origin/master/custom1.crt",
+                           "keyfile": "/etc/origin/master/custom1.key",
+                           "names": [ "public-master-host.com",
+                                      "other-master-host.com" ] },
+                         { "certfile": "/etc/origin/master/custom2.crt",
+                           "keyfile": "/etc/origin/master/custom2.key",
+                           "names": [ "some-hostname.com" ] }]
+        '''
+        if not issubclass(type(certificates), list):
+            raise errors.AnsibleFilterError("|failed expects certificates is a list")
+
+        if not issubclass(type(data_dir), unicode):
+            raise errors.AnsibleFilterError("|failed expects data_dir is unicode")
+
+        for certificate in certificates:
+            if not os.path.isfile(certificate['certfile']) and not os.path.isfile(certificate['keyfile']):
+                # Unable to find cert/key, try to prepend data_dir to paths
+                certificate['certfile'] = os.path.join(data_dir, certificate['certfile'])
+                certificate['keyfile'] = os.path.join(data_dir, certificate['keyfile'])
+                if not os.path.isfile(certificate['certfile']) and not os.path.isfile(certificate['keyfile']):
+                    # Unable to find cert/key in data_dir
+                    raise errors.AnsibleFilterError("|certificate and/or key does not exist %s, %s" %
+                                                    (certificate['certfile'], certificate['keyfile']))
+            try:
+                cert_txt = subprocess.check_output(["/usr/bin/openssl", "x509", "-text", "-noout",
+                                                    "-in", certificate['certfile']])
+            except subprocess.CalledProcessError:
+                raise errors.AnsibleFilterError("|failed to parse certificate %s" % certificate['certfile'])
+
+            certificate['names'] = []
+            if 'Subject Alternative Name' in cert_txt:
+                for match in re.finditer('(?<=DNS:)[a-zA-Z0-9.-]*', cert_txt):
+                    certificate['names'].append(match.group())
+            subject_line = re.search('Subject.*\n', cert_txt).group()
+            for match in re.finditer('(?<=CN=)[a-zA-Z0-9.-]*(?=/)', subject_line):
+                certificate['names'].append(match.group())
+            certificate['names'] = list(set(certificate['names']))
+        return certificates
```

Comment:
so we're auto-extracting the subject alt names and building the names list for them? if they provide a cert with a SAN that matches one of the internal DNS names, this would mean the infrastructure would get served a certificate it didn't recognize